### PR TITLE
unit_initial_queue.lua FIX duplicate msg to spec and ignoring remove msg

### PR DESCRIPTION
**Pull request by Helwor**
-Removed server msg sent with 's',
  spectator in any view mode receive allied message in pregame
-Code was faulty, would never receive remove message

### CODE REVIEW

File: LuaUI/Widgets/unit_initial_queue.lua [hunk 1]
```diff
@@ -398,16 +409,15 @@ function widget:RecvLuaMsg(msg, playerID)
...
 		elseif typeArg == 3 then
-			playerXBuildQueue[#playerXBuildQueue+1] = {unitDefID,x,y,z,face}
+			playerXBuildQueue[#playerXBuildQueue+1] = {unitDefIDorIndex,x,y,z,face}
 		elseif typeArg == 4 then
-			othersBuildQueue[teamID] = {{unitDefID,x,y,z,face}}
+			othersBuildQueue[teamID] = {{unitDefIDorIndex,x,y,z,face}}
 		end
 	end
 end
+end
```
Comment (sprunk) [inline LuaUI/Widgets/unit_initial_queue.lua:420]: > Error: Failed to load: unit_initial_queue.lua, [string "LuaUI/Widgets/unit_initial_queue.lua"]:420: '<eof>' expected near 'end'
```suggestion
```

File: LuaUI/Widgets/unit_initial_queue.lua [hunk 2]
```diff
@@ -381,14 +381,25 @@ function widget:RecvLuaMsg(msg, playerID)
 		--Header|unitdefID|x|y|z|facing
 		msg = msg:sub(4)
 		local msgArray = explode('|',msg)
-		local typeArg, unitDefID = tonumber(msgArray[1]), tonumber(msgArray[2])
+		local typeArg, unitDefIDorIndex = tonumber(msgArray[1]), tonumber(msgArray[2])
```
Comment (sprunk) [inline LuaUI/Widgets/unit_initial_queue.lua:384]: `unitDefIDorIndex` makes the code somewhat hard to read and to keep track of whether it's one or the other. Perhaps do it like this (conceptually and not necessarily verbatim)?
```lua
local typeArg = tonumber(msgArray[1])

if typeArg == 5 then
  -- blabla
  return
end

if typeArg == 2 then
  local index = tonumber(msgArray[2])
  -- blabla
  return
end

if typeArg == 1 or typeArg == 3 or typeArg == 4 then
  local unitDefID = tonumber(msgArray[2])
  -- rest as-is
end
```

**Review by sprunk — CHANGES_REQUESTED** [review]

**Comment by Helwor** [comment]
done

**Comment by sprunk** [comment]
tested, works as advertised. thanks!

**Pull request merged**